core/src/main/java/org/apache/accumulo/core/iterators/WrappingIterator.java

```diff
@@ -58,21 +58,21 @@ protected void setSource(SortedKeyValueIterator<Key,Value> source) {
   public Key getTopKey() {
     if (seenSeek == false)
       throw new IllegalStateException("never been seeked");
-    return getSource().getTopKey();
+    return source.getTopKey();
```

Comment:
In Sqrrl we use a LightWrappingIterator that does this. I'd be in favor of deprecating the WrappingIterator, but we might want to couple that with the bigger iterator rewrite project (ACCUMULO-3751).